OpenRA.Mods.Common/Traits/Conditions/GrantConditionOnDeploy.cs

```diff
@@ -183,7 +186,29 @@ Order IIssueDeployOrder.IssueDeployOrder(Actor self, bool queued)
 			return new Order("GrantConditionOnDeploy", self, queued);
 		}
 
-		bool IIssueDeployOrder.CanIssueDeployOrder(Actor self, bool queued) { return !IsTraitPaused && !IsTraitDisabled; }
+		bool IIssueDeployOrder.CanIssueDeployOrder(Actor self, bool queued)
+		{
+			if (IsTraitPaused || IsTraitDisabled || self.IsDead || self.Disposed)
+				return false;
+
+			if (queued || !Info.SmartDeploy || DeployState == DeployState.Undeployed || DeployState == DeployState.Undeploying)
+				return true;
+
+			foreach (var actor in self.World.Selection.Actors)
+			{
+				if (actor == self || actor.IsDead || !actor.IsInWorld)
+					continue;
+
+				var gcod = actor.TraitOrDefault<GrantConditionOnDeploy>();
+				if (gcod == null)
```

Comment:
I'm not sure if there's another part of the code that would prevent multiple GCOD rn (of course multiple enabled at the same time can't work, but different optional traits could be the case), but this could still better be `TraitsImplementing` along with a `!IsTraitDisabled` check.

Reply:
sure